testing/sdk_install.py

```diff
@@ -221,6 +208,37 @@ def uninstall(
             shakedown.pretty_duration(finish - cleanup_start),
             shakedown.pretty_duration(finish - start)))
 
+    state_summary = sdk_cmd.cluster_request('GET', '/mesos/state-summary').json()
+
+    # Sanity check: There should be no dangling resources in the state summary (DCOS-30314)
+    found_dangling_resources = False
+    service_role = '{}-role'.format(service_name)
+    for agent in state_summary['slaves']:
+        # resources should be grouped by role. check for any resources in our expected role:
+        matching_reserved_resources = agent['reserved_resources'].get(service_role, {})
+        if len(matching_reserved_resources) > 0:
+            log.error('Dangling resources on agent {}/{}: {}'.format(
+                agent['id'], agent['hostname'], matching_reserved_resources))
+            found_dangling_resources = True
+    if found_dangling_resources:
+        log.error(state_summary)
+        raise dcos.errors.DCOSException('Found dangling resources after uninstall of {}'.format(service_name))
+    log.info('No dangling resources for role {} were found'.format(service_role))
+
+    # Sanity check: There should be no framework entry for this service in the state summary (DCOS-29474)
+    dangling_frameworks = [fwk for fwk in state_summary['frameworks'] if fwk['name'] == service_name]
+    if len(dangling_frameworks) > 0:
+        log.error(state_summary)
+        raise dcos.errors.DCOSException('Found dangling frameworks after uninstall of {}'.format(service_name))
```

Comment:
Any reason for using a `DCOSException` here? Is it due to the retries around the uninstall functions? Remember that the `dcos` python package is being deprecated, and as such as regular Exception may be better here.

Reply:
Switched to `Exception` throughout. I think it was just to align with `_retried_install_impl()`, which calls into shakedown a ton and thereby could be returning `DCOSException`s in a lot of cases.